File: MappyC#/Mappy.cs
using System;
using Pluton;
using UnityEngine;

namespace Mappy
{
    public class Mappy : CSharpPlugin
    {
        public void On_PluginInit()
        {
            if (!Server.Loaded) return;
            DataStore.Flush("Mappy");
            IniParser ini = ConfigurationFile();
            if (ini.GetSetting("Settings", "enabled") == "1")
            {
                ServerConsoleCommands.Register("mappy").setCallback(GetCommand);
                string link = ini.GetSetting("Settings", "url");
                if (ini.GetSetting("Settings", "SendChat") == "1")
                {
                    DataStore.Add("Mappy", "SendChat", 1);
                    DataStore.Add("Mappy", "LinkChat", link + "chat.php");
                }
                else
                {
                    DataStore.Add("Mappy", "SendChat", 0);
                }
                DataStore.Add("Mappy", "Link", link + "server.php");
                DataStore.Add("Mappy", "LinkSize", link + "size.php");
                int mseconds = ToInt(ini.GetSetting("Settings", "Timer"));
                StartPlugin(mseconds);
            }
        }

        public void On_ServerInit()
        {
            DataStore.Flush("Mappy");
            IniParser ini = ConfigurationFile();
            if (ini.GetSetting("Settings", "enabled") == "1")
            {
                ServerConsoleCommands.Register("mappy").setCallback(GetCommand);
                string link = ini.GetSetting("Settings", "url");
                if (ini.GetSetting("Settings", "SendChat") == "1")
                {
                    DataStore.Add("Mappy", "SendChat", 1);
                    DataStore.Add("Mappy", "LinkChat", link + "chat.php");
                }
                else
                {
                    DataStore.Add("Mappy", "SendChat", 0);
                }
                DataStore.Add("Mappy", "Link", link + "server.php");
                DataStore.Add("Mappy", "LinkSize", link + "size.php");
                int mseconds = ToInt(ini.GetSetting("Settings", "Timer"));
                StartPlugin(mseconds);
            }
        }

        public void GetCommand(string[] args)
        {
            if (args[0] != "")
            {
                if (args[0] == "airdrop")
                {
                    if (args.Length >= 3)
                    {
                        string x = args[1];
                        string z = args[2];
                        World.AirDropAt(float.Parse(x), 0, float.Parse(z));
                    }
                }

                else if (args[0] == "kick")
                {
                    if (args.Length >= 2)
                    {
                        ulong sid = UInt64.Parse(args[1]);
                        if (Server.Players.ContainsKey(sid))
                        {
                            Pluton.Player player = Server.Players[sid];
                            string message = String.Join(" ", args);
                            message = String.Format("{0}", message.Replace(args[0] + " " + args[1] + " ", ""));
                            player.Kick(message);
                        }
                    }
                }

                else if (args[0] == "ban")
                {
                    if (args.Length >= 2)
                    {
                        ulong sid = UInt64.Parse(args[1]);
                        if (Server.Players.ContainsKey(sid))
                        {
                            Pluton.Player player = Server.Players[sid];
                            string message = String.Join(" ", args);
                            message = String.Format("{0}", message.Replace(args[0] + " " + args[1] + " ", ""));
                            player.Ban(message);
                        }
                    }
                }

                else if (args[0] == "message")
                {
                    if (args.Length >= 3)
                    {
                        ulong sid = UInt64.Parse(args[1]);
                        if (Server.Players.ContainsKey(sid))
                        {
                            Pluton.Player player = Server.Players[sid];
                            string message = String.Join(" ", args);
                            message = String.Format("{0}", message.Replace(args[0] + " " + args[1] + " ", ""));
                            player.Message(message);
                        }
                    }
                }

                else if (args[0] == "give")
                {
                    if (args.Length >= 4)
                    {
                        ulong sid = UInt64.Parse(args[1]);
                        if (Server.Players.ContainsKey(sid))
                        {
                            Pluton.Player player = Server.Players[sid];
                            int count = ToInt(args[2]);
                            int item = 0;
                            item = Pluton.InvItem.GetItemID(args[3]);
                            if (item == 0)
                                return;
                            player.Inventory.Add(item, count);
                        }
                    }
                }

                else if (args[0] == "teleport")
                {
                    if (args.Length >= 4)
                    {
                        ulong sid = UInt64.Parse(args[1]);
                        if (Server.Players.ContainsKey(sid))
                        {
                            Pluton.Player player = Server.Players[sid];
                            String x = args[2];
                            String z = args[3];
                            player.Teleport(float.Parse(x), World.GetGround(float.Parse(x), float.Parse(z)), float.Parse(z));
                        }
                    }
                }

                else if (args[0] == "animal")
                {
                    if (args.Length >= 4)
                    {
                        float x = float.Parse(args[1]);
                        float z = float.Parse(args[2]);
                        string animalname = args[3];
                        GameManager.server.CreateEntity("autospawn/animals/" + animalname, new UnityEngine.Vector3(x, World.GetGround(x, z), z)).Spawn(true);
                    }
                }

                else if (args[0] == "broadcast")
                {
                    if (args.Length >= 2)
                    {
                        string message = String.Join(" ", args);
                        message = String.Format("{0}", message.Replace(args[0] + " ", ""));
                        Server.Broadcast(message);
                    }
                }
            }
            else
            {
                Logger.Log("MAPPY PLUGIN & MAP CODE MADE BY Skully (SkullyDev)");
                Logger.Log("Proud member of Pluton-Team.ORG");
            }
        }

        private IniParser ConfigurationFile()
        {
            if (!Plugin.IniExists("ConfigurationFile"))
            {
                IniParser ini = Plugin.CreateIni("ConfigurationFile");
                ini.AddSetting("Settings", "enabled", "1");
                ini.AddSetting("Settings", "SendChat", "1");
                ini.AddSetting("Settings", "Timer", "5000");
                ini.AddSetting("Settings", "url", "http://www.example.com/mappy/");
                ini.Save();
            }
            return Plugin.GetIni("ConfigurationFile");
        }

        private void StartPlugin(int ms)
        {
            Plugin.CreateTimer("Send", ms).Start();
            string link = (string)DataStore.Get("Mappy", "LinkSize");
            string WorldSize = String.Format("&worldsize={0}", global::World.Size.ToString());
            Plugin.POST(link, WorldSize);
        }

        public void SendCallback(TimedEvent timer)
        {
            string post = String.Format("&time={0}&sleepers={1}", World.Time.ToString(), Server.SleepingPlayers.Count.ToString());
            if ((int)DataStore.Get("Mappy", "SendChat") == 1)
            {
                post = String.Format("{0}&showchat=true", post);
            }
            post = String.Format("{0}&players=::", post);
            foreach (Pluton.Player player in Server.ActivePlayers)
            {
                post = String.Format("{0};{1}:{2}:{3}:{4}", post, Uri.EscapeDataString(player.Name), player.X, player.Z, player.SteamID);
            }
            string link = (string)DataStore.Get("Mappy", "Link");
            try { Plugin.POST(link, post); } catch { }
        }

        public void On_Chat(Pluton.Events.ChatEvent Chat)
        {
            if ((int)DataStore.Get("Mappy", "SendChat") == 1)
            {
                Pluton.Player player = Chat.User;
                string post = String.Format("&chat={0}: {1}", Uri.EscapeDataString(player.Name), Uri.EscapeDataString(Chat.OriginalText));
                string link = (string)DataStore.Get("Mappy", "LinkChat");
                try { Plugin.POST(link, post); } catch { }
            }
        }

        private int ToInt(string obj)
        {
            int i;
            int.TryParse(obj, out i);
            return i;
        }
    }
}

File: MappyPY/Mappy.py
__author__ = 'Skully'
__version__ = '3.0'

import clr
import sys
import math

clr.AddReferenceByPartialName("UnityEngine")
clr.AddReferenceByPartialName("Pluton")
clr.AddReference("Assembly-CSharp")
import World as globalWorld
import UnityEngine
import Pluton
import System
from System import Uri

class Mappy:
    def ConfigurationFile(self):
        if not Plugin.IniExists("ConfigurationFile"):
            ini = Plugin.CreateIni("ConfigurationFile")
            ini.AddSetting("Settings", "enabled", "1")
            ini.AddSetting("Settings", "SendChat", "1")
            ini.AddSetting("Settings", "Timer", "60000")
            ini.AddSetting("Settings", "url", "http://www.example.com/mappy/")
            ini.Save()
        return Plugin.GetIni("ConfigurationFile")

    def On_PluginInit(self):
        DataStore.Flush("Mappy")
        ini = self.ConfigurationFile()
        if ini.GetSetting("Settings", "enabled") == "1":
            ServerConsoleCommands.Register("mappy").setCallback("ReceiveCommand")
            link = ini.GetSetting("Settings", "url")
            if ini.GetSetting("Settings", "SendChat") == "1":
                DataStore.Add("Mappy", "SendChat", 1)
                DataStore.Add("Mappy", "LinkChat", link + "chat.php")
            DataStore.Add("Mappy", "Link", link + "server.php")
            DataStore.Add("Mappy", "LinkSize", link + "size.php")
            mseconds = ini.GetSetting("Settings", "Timer")
            msec = int(mseconds)
            Plugin.CreateTimer("SendSizeOnce", 15000).Start()
            Plugin.CreateTimer("Send", msec).Start()

    def ReceiveCommand(self, args):
        if args[0] == "":
            return
        if args[0] == "airdrop":
            x = args[1]
            z = args[2]
            World.AirDropAt(float(x), 0, float(z))
        elif args[0] == "kick":
            steamid = int(args[1])
            player = Pluton.Players[steamid]
            player.Kick("Mappy control panel kick")
        elif args[0] == "message":
            steamid = int(args[1])
            player = Pluton.Players[steamid]
            message = str.Join(" ", args)
            message = message.replace(args[0] + " " + args[1] + " ", "")
            player.Message(message)
        elif args[0] == "teleport":
            x = args[2]
            z = args[3]
            steamid = int(args[1])
            player = Pluton.Players[steamid]
            player.Teleport(float(x), World.GetGround(float(x), float(z)), float(z))
        elif args[0] == "give":
            steamid = int(args[1])
            player = Pluton.Players[steamid]
            count = args[2]
            # GIVE CODE USED FROM "Give" balu92's PLUGIN
            item = Pluton.InvItem.GetItemID(args[3])
            player.Inventory.Add(item, int(count))
        elif args[0] == "animal":
            x = args[1]
            z = args[2]
            animalname = args[3]
            World.SpawnAnimal(animalname, float(x), float(z))
        elif args[0] == "broadcast":
            message = str.Join(" ", args)
            message = message.replace(args[0] + " ", "")
            Server.Broadcast(message)

    def SendSizeOnceCallback(self, timer):
        timer.Kill()
        DataStore.Add("Mappy", "SizeSent", 1)
        link = DataStore.Get("Mappy", "LinkSize")
        WorldSize = "&worldsize=" + str(globalWorld.Size)
        Plugin.POST(link, WorldSize)

    def SendCallback(self, timer):
        ServersTime = str(World.Time)
        post = "&time=" + ServersTime + "&sleepers=" + str(Server.SleepingPlayers.Count)
        if DataStore.Get("Mappy", "SendChat") == 1:
            post = post + "&showchat=true"
        post = post + "&players=::"
        for player in Server.ActivePlayers:
            if player.Location:
                Name = Uri.EscapeDataString(player.Name)
                coords = str(player.Location)
                stripit = str.strip(coords, "()")
                sripped = str.strip(stripit, ",")
                splitted = sripped.split(" ")
                coordx = splitted[0]
                coordz = splitted[2]
                steamid = player.SteamID
                post = post + ";" + Name + ":" + coordx + ":" + coordz + ":" + steamid
        link = DataStore.Get("Mappy", "Link")
        Plugin.POST(link, post)

    def On_Chat(self, Chat):
        if DataStore.Get("Mappy", "SendChat") == 1:
            Message = Uri.EscapeDataString(Chat.OriginalText)
            Sender = Uri.EscapeDataString(Chat.User.Name)
            link = DataStore.Get("Mappy", "LinkChat")
            post = "&chat=" + Sender + ": " + Message
            Plugin.POST(link, post)
